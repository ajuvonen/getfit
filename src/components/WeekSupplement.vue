<script setup lang="ts">
import {useI18n} from 'vue-i18n';
import type {Training} from '@/types';
import useWeekDays from '@/hooks/weekdays';
import {getIcon} from '@/utils';

defineProps<{
  trainings: Training[];
  dayIndex: number;
}>();

const {t} = useI18n();
const {weekdays} = useWeekDays();
</script>
<template>
  <div v-if="trainings.length">
    <div class="mt-4 text-subtitle-2 text-uppercase">
      {{ weekdays[dayIndex] }}
    </div>
    <ul>
      <li
        v-for="training in trainings"
        :key="training.id"
        class="mt-1 week-supplement__instructions text-body-1"
      >
        <span>
          <v-icon
            :icon="getIcon(training.activity)"
            :title="t(`activities.${training.activity}`)"
            :aria-label="t(`activities.${training.activity}`)"
          />
          {{ training.title || t(`activities.${training.activity}`) }}:&nbsp;</span
        >
        <span v-if="training.instructions">{{ training.instructions }}</span>
        <i v-else>{{ t('weekSupplement.noInstruction') }}</i>
      </li>
    </ul>
  </div>
</template>
<style lang="scss" scoped>
ul {
  list-style-type: none;
}

.v-icon {
  vertical-align: bottom;
}

.week-supplement__instruction {
  padding-left: 1.8em;
  text-indent: -1.8em;
}
</style>
