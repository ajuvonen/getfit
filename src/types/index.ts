export type Training = {
  id: string;
  weekId: string;
  dayIndex: number;
  activity: string;
  title: string;
  description: string;
  duration: number;
  intensity: Intensity;
  location: string;
};

export type Week = {
  id: string;
  trainings: Training[];
};

export type ScheduleSettings = {
  name: string;
  startsOnSunday: boolean;
  startDate: Date | null;
  actualWeekNumbering: boolean;
  availableActivities: string[];
  defaultStartTime: {hours: number; minutes: number; seconds: number};
  defaultDuration: number;
  unitOfTime: 'm' | 'h';
};

export enum Intensity {
  'LIGHT',
  'NORMAL',
  'MEDIUM',
  'HEAVY',
}

export type BaseActivity = {
  value: string;
  icon: string;
};

export type LocalizedActivity = BaseActivity & {
  title: string;
};

export type CalendarEvent = {
  title: string;
  description: string;
  start: [number, number, number, number, number];
  duration: {minutes: number};
  location: string;
  categories: string[];
  status: 'CONFIRMED';
  busyStatus: 'BUSY';
  transp: 'OPAQUE';
  productId: 'ajuvonen/getfit';
  classification: 'PRIVATE';
  calname: string;
};
