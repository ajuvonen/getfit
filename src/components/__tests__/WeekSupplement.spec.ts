import {mount} from '@vue/test-utils';
import {describe, it, expect} from 'vitest';
import {v4 as uuidv4} from 'uuid';
import {Intensity, type Training} from '@/types';
import WeekSupplement from '@/components/WeekSupplement.vue';

const trainings: Training[] = [
  {
    id: uuidv4(),
    weekId: uuidv4(),
    dayIndex: 0,
    activity: 'boxing',
    title: '',
    description: 'Sparring at the gym',
    duration: 1,
    intensity: Intensity.MEDIUM,
    location: 'Total wreck gym',
  },
];

describe('WeekSupplement', () => {
  it('mounts', () => {
    const wrapper = mount(WeekSupplement, {
      props: {
        trainings,
        dayIndex: 0,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
