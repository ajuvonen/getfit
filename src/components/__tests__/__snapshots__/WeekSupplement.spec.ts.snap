// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WeekSupplement > mounts 1`] = `
"<div data-v-22de64f8="">
  <div data-v-22de64f8="" class="mt-4 text-subtitle-2 text-uppercase">Monday</div>
  <ul data-v-22de64f8="">
    <li data-v-22de64f8="" class="mt-1 week-supplement__instruction text-body-1"><span data-v-22de64f8=""><i data-v-22de64f8="" class="mdi-boxing-glove mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true" title="Boxing" aria-label="Boxing"></i> Boxing:&nbsp;</span><span data-v-22de64f8="">Sparring at the gym</span></li>
  </ul>
</div>"
`;
